src/main/kotlin/org/opensearch/replication/task/index/IndexReplicationTask.kt

```diff
@@ -461,7 +461,7 @@ open class IndexReplicationTask(id: Long, type: String, action: String, descript
                             continue
                         }
                         val setting = indexScopedSettings[key]
-                        if (!setting.isPrivateIndex) {
+                        if (!setting.isPrivateIndex && !setting.isFinal) {
```

Comment:
- Skipping final settings is good but i think we should still explicitly block the `index.replication.type` setting. Probably we should do both changes as excluding final settings might be helpful in other corner cases as well.  
- For skipping final settings, I think we can just filter out the final settings while initializing `settingsList`(similar to blocklisted settings).

Reply:
As discussed offline, lets keep as it is.